src/main/java/babblebot/task/Task.java

```diff
@@ -0,0 +1,86 @@
+package babblebot.task;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * It serves as the base class for more specific types of tasks such as Todo, Deadline, and Event.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task with the specified description.
+     * The task is initially marked as not done.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of the task.
+     * "X" if the task is done, " " if the task is not done.
+     *
+     * @return The status icon of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsUndone() {
+        isDone = false;
+    }
+
+    /**
+     * Converts the Task to a format suitable for file storage.
+     *
+     * @return A string representation of the Task in file format.
+     */
+    public String toFileFormat() {
+        if (getStatusIcon().equals("X")) {
+            return "1" + " | " + this.description;
+        } else {
+            return "0" + " | " + this.description;
+        }
+    }
+
+    /**
+     * Returns the presence of the given keyword within the description.
+     * "true" if the keyword is present, "false" if the keyword is not.
+     *
+     * @param keyword The supplied keyword string.
+     * @return The boolean indicating presence of the keyword
+     */
+    public boolean hasKeyword(String keyword) {
+        String[] descriptionArray = description.split(" ");
+        for (String word : descriptionArray) {
+            if (word.equals(keyword)) {
+                return true;
+            }
+        }
+        return false;
+
```

Comment:
Its hard to find issues with your code quality, but perhaps you did not mean to add an extra newline between the return statement and the closing brace 😆